plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -170,15 +170,15 @@ public void setPropertyValue(String path, Object value) {
     }
 
     public String getIndex() {
```

Comment:
sure I think what I have in mind apart from _index, _type and _id are cases where despite you are using proper metadata fields you don't have them in the document because they are disabled in the mapping.